src/condor_tests/test_multifile_curl_plugin_timeout.py

```diff
@@ -0,0 +1,207 @@
+        #Not enough xfer, timeout (20:30)
+        lambda: 'n' * 20,
+        lambda: "Delay=30",
+        lambda: ONE_BYTE_STR,
+    ),
+}
+
+#--------------------------------------------------------------------------
+#Internal Iterator class to send xfer data correctly
+class SlowIterator:
+    def __iter__(self):
+        for data in (l() for l in self.res):
+            #print(data)
+            if "Delay" in data:
+                time = data.split("=")
+                sleep(int(time[1]))
+                continue
+            else:
+                yield data
+    def __init__(self,response):
+        self.res = response
+
+#Handler fuction to return response of iterator object data
+def req_handler(request, response):
+    return Response(SlowIterator(response))
+
+TEST_NUM = -1
+#Server handling of an http request
+def server_data_xfer(srv,response,key):
+    global TEST_NUM
+    TEST_NUM += 1
+    srv.expect_request(f"/testserver/{TEST_NUM}/{key}").respond_with_handler(lambda r: req_handler(r,response))
+    return "http://localhost:{0}/testserver/{1}/{2}".format(srv.port,TEST_NUM,key)
+
+#--------------------------------------------------------------------------
+#Submit all the test cases as job
+SERVERS = {} #List of servers (one per test)
+@action
+def submit_jobs(default_condor,test_dir,path_to_sleep):
+    global SERVERS
+    job_handles = {}
+    #For each test submit the job
+    for key in TEST_CASES.keys():
+        #Make a pytest HTTPServer
+        server = HTTPServer()
+        #Start server
+        if not server.is_running():
+            server.start()
+        #Add server to list of servers
+        port = f"{server.port}"
+        SERVERS[port] = server
+        #Submit job
+        job = default_condor.submit(
+            {
+                "executable": path_to_sleep,
+                "arguments": "1",
+                "log": (test_dir / "ran.log").as_posix(),
+                "transfer_input_files": server_data_xfer(server,TEST_CASES[key],key),
+                "should_transfer_files": "YES",
+            }
+        )
+        #Add job to job_handle dictionaries
+        job_handles[key] = (port,job)
+    yield job_handles
+
+#--------------------------------------------------------------------------
+@action(params={name: name for name in TEST_CASES})
+def job_info(request, submit_jobs):
+    return (request.param, submit_jobs[request.param][1],submit_jobs[request.param][0])
+@action
+#Get Job Name
+def job_name(job_info):
+    return job_info[0]
+#Get job handle
+@action
+def job_handle(job_info):
+    return job_info[1]
+#Get server port that job is running on
+@action
+def job_server_port(job_info):
+    return job_info[2]
+#Wait for the job to finish
+@action
+def job_wait(job_handle,job_server_port):
+    job_handle.wait(condition=ClusterState.all_terminal,timeout=80)
+    #Stop server if running
+    if SERVERS[job_server_port].is_running:
+        SERVERS[job_server_port].stop()
+    return job_handle
```

Comment:
I had forgotten that this would be necessary, so good catch.  You could have avoided another global here by passing the server object itself along instead of its port number in `job_handles`.  (Assuming the server object knows its own port number; if it doesn't, you can pass both along.)

Good to change if you have time but not necessary.